src/functions.php

```diff
@@ -291,18 +293,44 @@ function default_http_handler()
  */
 function default_user_agent()
 {
-    $version = (string) ClientInterface::VERSION;
-    if ($version[0] === '5') {
-        return \GuzzleHttp\Client::getDefaultUserAgent();
+    $version = guzzle_major_version();
+    // If Guzzle 6 or 7 installed
+    if ($version === 6 || $version === 7) {
+        return \GuzzleHttp\default_user_agent();
     }
 
-    if ($version[0] === '6') {
-        return \GuzzleHttp\default_user_agent();
+    // If Guzzle 5 installed
+    if ($version === 5) {
+        return \GuzzleHttp\Client::getDefaultUserAgent();
     }
 
     throw new \RuntimeException('Unknown Guzzle version: ' . $version);
 }
 
+/**
+ * Get the major version of guzzle that is installed.
+ *
+ * @internal This function is internal and should not be used outside aws/aws-sdk-php.
+ * @return int
+ * @throws \RuntimeException
+ */
+function guzzle_major_version()
+{
```

Comment:
Maybe keep a static variable cache of the result, within this function body?

Reply:
I think this makes sense, esp considering how often this method will be called.